nixos/modules/tasks/network-interfaces-systemd.nix

```diff
@@ -53,20 +59,27 @@ in
     systemd.network =
       let
         domains = cfg.search ++ (optional (cfg.domain != null) cfg.domain);
-        genericNetwork = override:
-          let gateway = optional (cfg.defaultGateway != null) cfg.defaultGateway.address
+        defaultGateways = optional (cfg.defaultGateway != null) cfg.defaultGateway.address
             ++ optional (cfg.defaultGateway6 != null) cfg.defaultGateway6.address;
-          in {
+        genericNetwork = override:
+          {
             DHCP = override (dhcpStr cfg.useDHCP);
-          } // optionalAttrs (gateway != [ ]) {
-            gateway = override gateway;
+          } // optionalAttrs (defaultGateways != [ ]) {
+            gateway = override defaultGateways;
           } // optionalAttrs (domains != [ ]) {
             domains = override domains;
+          } // optionalAttrs (! cfg.enableIPv6) {
+            # if IPv6 is disabled, we shoud define additionnal attributes
+            # see https://github.com/coreos/bugs/issues/1419
+            networkConfig.LinkLocalAddressing = "no";
```

Comment:
I am not a fan of this... Do you really have to do this?

The way I read the bug report is:
If you disable IPv6 (which would only be configured by SLAAC anyway, so your network already is opted-in?) then booting up causes issues?
Why would you disable IPv6 then? I mean there is no good reason to NOT support IPv6 in 2018... Yes corporate networks and stuff but I am not sorry.. I has been many years to upgrade your legacy IT…

Reply:
I use my nixos box in a home gateway scenario. My provider does not provide a static v6 address, which is why I didn't configure IPv6 in my home (I was waiting for time to devise a solution for this).

Anyways, I think if ipv6 is actually disable (`cfg.enableIPv6 = false`), then we should indeed disable those options, for coherence sake, and to avoid interfaces getting stuck in the `configuring` stage (and thus the `network-online.target` failing).

Reply:
I agree. If someone decides to disable IPv6 we should honor that.

Are you able to write a small NixOS test to verify that the behavior is as expected? (Both with IPv6 disabled/enabled)

Reply:
I'll think about a test, do you have any pointers for writing one ?

Also, I need to fix the `error: The unique option `systemd.network.networks.40-pLAN.networkConfig.IPv6AcceptRA' is defined multiple times, in `/var/nixpkgs/nixos/modules/tasks/network-interfaces-systemd.nix' and `/var/nixpkgs/nixos/modules/tasks/network-interfaces-systemd.nix'.
` error I am having with the current implementation (due to the fact that, for vlans, the underlying physical interface is referred to multiple times). Do you know anyone would could help me fix this ?

Thanks